File: frontend/temporal-trace-ui/src/app/models/task-work-update.ts
export interface TaskWorkUpdate {
  id: number;
  taskId: number;
  note: string;
  statusAfter: string | null;
  minutesSpent: number | null;
  createdAt: string;
}

export interface CreateTaskWorkUpdateRequest {
  note: string;
  statusAfter: string | null;
  minutesSpent: number | null;
}

File: backend/TemporalTrace.Api/Models/TaskWorkUpdate.cs
namespace TemporalTrace.Api.Models;

public class TaskWorkUpdate
{
    public int Id { get; set; }
    public int TaskId { get; set; }
    public string Note { get; set; } = string.Empty;
    public string? StatusAfter { get; set; }
    public int? MinutesSpent { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
